design/manual-authn-and-authz/README.md

```diff
@@ -0,0 +1,65 @@
+# Manual Authentication and Authorization workaround
+
+* Status: **Proposal**
+* Discussion: [GitHub PR](https://github.com/strimzi/strimzi-kafka-operator/pull/599)
+
+## Motivation
+
+The long term plan for Strimzi is to have custom resources to define Users for Authentication and Authorization rules.
+The custom resources should be read by operator style application which would be responsible for the setup of the users or ACL rules.
+Until such operator (User Operator - UO) is available, it would be useful to have a way how to setup Authentication and Authorization at least in manual / semi-manual way.
+This proposal covers such workaround.
+There is no long term commitment to support the mechanisms suggested in this proposal.
+After the final implementation of the operator for Authorization and Authentication is implemented, it will not be possible anymore to use this workaround.
+
+## Authentication
+
+The Cluster Operator (CO) maintains a _Clients CA_. 
+This Client CA is set as truststore for the TLS enabled Kafka interface which is designated for client connections (port 9093).
+Currently, this _Clients CA_ does not enforce client authentication.
+And enabling such configuration through `spec.kafka.config` is not possible because the required options are in the _forbidden options_ list.
+This proposal suggests to modify the list of forbidden options to allow end-users to enable or disable the TLS client authentication.
+This change will be only temporary and will be removed once the actual UO is in place with the needed configuration.
+
+User who want to use manual setup of authentication will be able to do so in the following steps:
+
+* Modify the Kafka cluster resource and add configuration options to enable TLS client authentication for the clients port (option `listener.name.clienttls.ssl.client.auth=required`)
+* Use the Clients CA to manually issue user certificates. 
+  The CA public and private key can be downloaded from the Kubernetes / OpenShift secret and the user certificates can be signed separately. 
+  The CA private key should be kept secret.
```

Comment:
Today we have `my-cluster-kafka-cert` containingjust the `clients-ca.crt` which is the way the customer can get the Clients CA for validating brokers certs signed with that (so just for server auth). Then we have `my-cluster-kafka-clients-ca` which contains both `clients-ca.crt` and `clients-ca.key` (which is used today for signing broker certs used on clients communication). Now because the `clients-ca.key` is going to be used for signing clients certs as well (for clients auth) so clients key and cert are both useful, should we remove the `my-cluster-kafka-cert` secret or we want to leave that one because this solution is temporary?

Reply:
I'm not sure there will be any use for `my-cluster-kafka-cert`. Maybe we could use it to set the truststore for the `clienttls` listener.